src/recipe/commands/handlers/deleteRecipe.handler.ts

```diff
@@ -0,0 +1,20 @@
+import { CommandHandler, ICommandHandler } from '@nestjs/cqrs';
+import { FileService } from '../../../file/file.service';
+import { DeleteRecipeCommand } from '../impl/deleteRecipe.command';
+import { RecipeDAO } from '../../recipe.dao';
+
+@CommandHandler(DeleteRecipeCommand)
+export class DeleteRecipeHandler
+  implements ICommandHandler<DeleteRecipeCommand>
+{
+  constructor(
+    private readonly recipeDAO: RecipeDAO,
+    private readonly fileService: FileService,
+  ) {}
+  async execute(command: DeleteRecipeCommand) {
+    const { recipeId } = command;
+    const recipeToDelete = await this.recipeDAO.findRecipeById(recipeId);
```

Comment:
what will happen when recipe in not found?

Reply:
It will throw NotFoundException, method findRecipeById from recipeDAO throws this exception when recipe is not found, that is why i call it here